### ISSUE TITLE
Rescan resets "Set as preview"

### ISSUE THREAD

**Issue by lloydpick** [Floppy/van_dam#501]
I have a large intricate model which gets picked up as the default preview, so I manually change it to a different model, but when re-scanning, this gets lost and reset.

**Comment by lloydpick** [comment]
I just checked this with the latest image, footer says `v0.19.0 (2e7d1324)`, and it's still resetting the preview after Rescan.

Log file...

```
I, [2022-06-27T17:31:49.073561 #17]  INFO -- : [b52e8be5-5f0a-41ed-b2f7-a97dd7481c84] [ActiveJob] Enqueued LibraryScanJob (Job ID: 92e6452f-4098-4f2b-b063-408da16c563d) to Async(default) with arguments: #<GlobalID:0x00007fed2fc7c5f8 @uri=#<URI::GID gid://van-dam/Library/6>>
I, [2022-06-27T17:31:49.074653 #17]  INFO -- : [b52e8be5-5f0a-41ed-b2f7-a97dd7481c84] method=PATCH path=/libraries/6 format=html controller=LibrariesController action=update status=302 duration=6.93 view=0.00 db=2.20 location=http://192.168.1.88:3214/libraries/6
I, [2022-06-27T17:31:49.076624 #17]  INFO -- : [ActiveJob] [LibraryScanJob] [92e6452f-4098-4f2b-b063-408da16c563d] Performing LibraryScanJob (Job ID: 92e6452f-4098-4f2b-b063-408da16c563d) from Async(default) enqueued at 2022-06-27T17:31:49Z with arguments: #<GlobalID:0x00007fed2fd09a20 @uri=#<URI::GID gid://van-dam/Library/6>>
I, [2022-06-27T17:31:49.087102 #17]  INFO -- : [ActiveJob] [LibraryScanJob] [92e6452f-4098-4f2b-b063-408da16c563d] Enqueued ModelScanJob (Job ID: 4afbb045-66e2-472a-9245-1e7b772c9b64) to Async(default) with arguments: #<GlobalID:0x00007fed2fd36c50 @uri=#<URI::GID gid://van-dam/Model/26>>
I, [2022-06-27T17:31:49.090267 #17]  INFO -- : [ActiveJob] [ModelScanJob] [4afbb045-66e2-472a-9245-1e7b772c9b64] Performing ModelScanJob (Job ID: 4afbb045-66e2-472a-9245-1e7b772c9b64) from Async(default) enqueued at 2022-06-27T17:31:49Z with arguments: #<GlobalID:0x00007fed2fa0bc60 @uri=#<URI::GID gid://van-dam/Model/26>>
I, [2022-06-27T17:31:49.093247 #17]  INFO -- : [ActiveJob] [LibraryScanJob] [92e6452f-4098-4f2b-b063-408da16c563d] Enqueued ModelScanJob (Job ID: 3b16375e-70b8-4c5f-9f74-f73b1a35cb1d) to Async(default) with arguments: #<GlobalID:0x00007fed2f9c9360 @uri=#<URI::GID gid://van-dam/Model/33>>
E, [2022-06-27T17:31:49.099862 #17] ERROR -- : [ActiveJob] [LibraryScanJob] [92e6452f-4098-4f2b-b063-408da16c563d] #<Model id: nil, name: "Humiditysensor", path: "/humiditysensor", library_id: 6, created_at: nil, updated_at: nil, preview_file_id: nil, creator_id: nil, scale_factor: 0.1e3, tag_list: nil>
E, [2022-06-27T17:31:49.100810 #17] ERROR -- : [ActiveJob] [LibraryScanJob] [92e6452f-4098-4f2b-b063-408da16c563d] ["Path has already been taken"]
I, [2022-06-27T17:31:49.105294 #17]  INFO -- : [ActiveJob] [LibraryScanJob] [92e6452f-4098-4f2b-b063-408da16c563d] Enqueued ModelScanJob (Job ID: f67e3178-de77-485c-b7d2-9beafc2b1a1f) to Async(default) with arguments: #<GlobalID:0x00007fed2f3e0458 @uri=#<URI::GID gid://van-dam/Model/28>>
E, [2022-06-27T17:31:49.114222 #17] ERROR -- : [ActiveJob] [LibraryScanJob] [92e6452f-4098-4f2b-b063-408da16c563d] #<Model id: nil, name: "Printeraccessories", path: "/printeraccessories", library_id: 6, created_at: nil, updated_at: nil, preview_file_id: nil, creator_id: nil, scale_factor: 0.1e3, tag_list: nil>
E, [2022-06-27T17:31:49.114973 #17] ERROR -- : [ActiveJob] [LibraryScanJob] [92e6452f-4098-4f2b-b063-408da16c563d] ["Path has already been taken"]
I, [2022-06-27T17:31:49.122692 #17]  INFO -- : [ActiveJob] [LibraryScanJob] [92e6452f-4098-4f2b-b063-408da16c563d] Enqueued ModelScanJob (Job ID: 41da578f-54a1-4c10-bb5b-c53fcc3f452b) to Async(default) with arguments: #<GlobalID:0x00007fed2eeee4e0 @uri=#<URI::GID gid://van-dam/Model/30>>
I, [2022-06-27T17:31:49.132573 #17]  INFO -- : [ActiveJob] [LibraryScanJob] [92e6452f-4098-4f2b-b063-408da16c563d] Enqueued ModelScanJob (Job ID: ea653c8b-031a-487c-afb8-e7d9089fce14) to Async(default) with arguments: #<GlobalID:0x00007fed2ed91bb0 @uri=#<URI::GID gid://van-dam/Model/31>>
I, [2022-06-27T17:31:49.139149 #17]  INFO -- : [ActiveJob] [ModelScanJob] [4afbb045-66e2-472a-9245-1e7b772c9b64] Performed ModelScanJob (Job ID: 4afbb045-66e2-472a-9245-1e7b772c9b64) from Async(default) in 51.72ms
I, [2022-06-27T17:31:49.140702 #17]  INFO -- : [eabc23c7-b4a9-4aa2-869e-3601dfdf4d53] method=GET path=/libraries/6 format=html controller=LibrariesController action=show status=200 duration=61.34 view=16.92 db=34.06
I, [2022-06-27T17:31:49.143958 #17]  INFO -- : [ActiveJob] [ModelScanJob] [3b16375e-70b8-4c5f-9f74-f73b1a35cb1d] Performing ModelScanJob (Job ID: 3b16375e-70b8-4c5f-9f74-f73b1a35cb1d) from Async(default) enqueued at 2022-06-27T17:31:49Z with arguments: #<GlobalID:0x00007fed2ee18f70 @uri=#<URI::GID gid://van-dam/Model/33>>
E, [2022-06-27T17:31:49.144886 #17] ERROR -- : [ActiveJob] [LibraryScanJob] [92e6452f-4098-4f2b-b063-408da16c563d] #<Model id: nil, name: "Temptowers", path: "/temptowers", library_id: 6, created_at: nil, updated_at: nil, preview_file_id: nil, creator_id: nil, scale_factor: 0.1e3, tag_list: nil>
E, [2022-06-27T17:31:49.145697 #17] ERROR -- : [ActiveJob] [LibraryScanJob] [92e6452f-4098-4f2b-b063-408da16c563d] ["Path has already been taken"]
I, [2022-06-27T17:31:49.148737 #17]  INFO -- : [ActiveJob] [LibraryScanJob] [92e6452f-4098-4f2b-b063-408da16c563d] Enqueued ModelScanJob (Job ID: fe159d19-f5f5-47db-afcb-e6de218ad4c8) to Async(default) with arguments: #<GlobalID:0x00007fed2edef008 @uri=#<URI::GID gid://van-dam/Model/34>>
I, [2022-06-27T17:31:49.150519 #17]  INFO -- : [ActiveJob] [LibraryScanJob] [92e6452f-4098-4f2b-b063-408da16c563d] Performed LibraryScanJob (Job ID: 92e6452f-4098-4f2b-b063-408da16c563d) from Async(default) in 76.66ms
I, [2022-06-27T17:31:49.152906 #17]  INFO -- : [ActiveJob] [ModelScanJob] [f67e3178-de77-485c-b7d2-9beafc2b1a1f] Performing ModelScanJob (Job ID: f67e3178-de77-485c-b7d2-9beafc2b1a1f) from Async(default) enqueued at 2022-06-27T17:31:49Z with arguments: #<GlobalID:0x00007fed2ee49cd8 @uri=#<URI::GID gid://van-dam/Model/28>>
I, [2022-06-27T17:31:49.167733 #17]  INFO -- : [ActiveJob] [ModelScanJob] [3b16375e-70b8-4c5f-9f74-f73b1a35cb1d] Performed ModelScanJob (Job ID: 3b16375e-70b8-4c5f-9f74-f73b1a35cb1d) from Async(default) in 28.08ms
I, [2022-06-27T17:31:49.169800 #17]  INFO -- : [ActiveJob] [ModelScanJob] [41da578f-54a1-4c10-bb5b-c53fcc3f452b] Performing ModelScanJob (Job ID: 41da578f-54a1-4c10-bb5b-c53fcc3f452b) from Async(default) enqueued at 2022-06-27T17:31:49Z with arguments: #<GlobalID:0x00007fed2d2c74b0 @uri=#<URI::GID gid://van-dam/Model/30>>
I, [2022-06-27T17:31:49.194434 #17]  INFO -- : [ActiveJob] [ModelScanJob] [41da578f-54a1-4c10-bb5b-c53fcc3f452b] Performed ModelScanJob (Job ID: 41da578f-54a1-4c10-bb5b-c53fcc3f452b) from Async(default) in 26.34ms
I, [2022-06-27T17:31:49.196395 #17]  INFO -- : [ActiveJob] [ModelScanJob] [ea653c8b-031a-487c-afb8-e7d9089fce14] Performing ModelScanJob (Job ID: ea653c8b-031a-487c-afb8-e7d9089fce14) from Async(default) enqueued at 2022-06-27T17:31:49Z with arguments: #<GlobalID:0x00007fed2cf1eae8 @uri=#<URI::GID gid://van-dam/Model/31>>
I, [2022-06-27T17:31:49.198630 #17]  INFO -- : [ActiveJob] [ModelScanJob] [f67e3178-de77-485c-b7d2-9beafc2b1a1f] Performed ModelScanJob (Job ID: f67e3178-de77-485c-b7d2-9beafc2b1a1f) from Async(default) in 47.66ms
I, [2022-06-27T17:31:49.200687 #17]  INFO -- : [ActiveJob] [ModelScanJob] [fe159d19-f5f5-47db-afcb-e6de218ad4c8] Performing ModelScanJob (Job ID: fe159d19-f5f5-47db-afcb-e6de218ad4c8) from Async(default) enqueued at 2022-06-27T17:31:49Z with arguments: #<GlobalID:0x00007fed2cf7a050 @uri=#<URI::GID gid://van-dam/Model/34>>
```

**Comment by Floppy** [comment]
This fix went into 0.20.0, which might have still being building and publishing when you checked `latest`. Should be there now though, give it a go?

**Comment by lloydpick** [comment]
Yup, this is fixed now
